arviz/plots/dotplot.py

```diff
@@ -0,0 +1,153 @@
+import math
+import numpy as np
+
+from ..rcparams import rcParams
+from .plot_utils import get_plotting_function
+
+
+def plot_dot(
+    values=None,
+    binwidth=None,
+    dotsize=1,
+    stackratio=1,
+    hdi_prob=None,
+    rotated=False,
+    dotcolor="C0",
+    intervalcolor="C3",
+    markersize=None,
+    figsize=None,
+    linewidth=None,
+    point_estimate="auto",
+    quantiles=50,
+    quartiles = True,
+    point_interval=None,
+    plot_kwargs=None,
+    interval_kwargs=None,
+    backend=None,
+    backend_kwargs=None,
+    **kwargs
+):
+
+    """Plot distribution as dot plot or quantile dot plot.
+
+    This function uses the Wilkinson's Algorithm
+    (https://www.cs.uic.edu/~wilkinson/Publications/dotplots.pdf)to allot dots to bins.
+    The quantile dot plots was inspired from the paper(When (ish) is My Bus?:
+    User-centered Visualizations of Uncertainty in Everyday).
+
+    Parameters
+    ----------
+    values : array-like
+        Values to plot
+    binwidth : float
+        Width of the bin for drawing the dot plot.
+    dotsize : float
+        The size of the dots relative to the bin width. The default, 1, makes dots be
+        just about as wide as the bin width.
+    stackratio : float
+        The distance between the center of the dots in the same stack relative to the bin height.
+        The default, 1, makes dots in the same stack just touch each other.
+    point_interval : bool
+        Plots the point interval. Uses hdi_prob to plot the HDI interval
+    point_estimate : Optional[str]
+        Plot point estimate per variable. Values should be ‘mean’, ‘median’, ‘mode’ or None.
+        Defaults to ‘auto’ i.e. it falls back to default set in rcParams.
+    dotcolor : string
+        The color of the dots
+    intervalcolor : string
+        The color of the interval
+    linewidth : int
+        Line width throughout. If None it will be autoscaled based on figsize.
+    markersize : int
+        Markersize throughout. If None it will be autoscaled based on figsize.
+    hdi_prob : float
+        Valid only when point_interval is True. Plots HDI for chosen percentage of density.
+        Defaults to 0.94.
+    rotated : bool
+        Whether to rotate the dot plot by 90 degrees.
+    quantiles : int
```

Comment:
`plot_kde` takes `quantiles` as a list of quantiles to plot. I wonder if we should use a different kwarg name than `quantiles` here for consistency. Perhaps `nquantiles` or `numquantiles`, which would also be more descriptive?

Reply:
Going with `nquantiles`